components/wallet-logger.js

```diff
@@ -0,0 +1,321 @@
+import { useRouter } from 'next/router'
+import LogMessage from './log-message'
+import { createContext, useCallback, useContext, useEffect, useMemo, useRef, useState } from 'react'
+import { Checkbox, Form } from './form'
+import { useField } from 'formik'
+import styles from '@/styles/log.module.css'
+import { Button } from 'react-bootstrap'
+import { useToast } from './toast'
+import { useShowModal } from './modal'
+import { WALLET_LOGS } from '@/fragments/wallet'
+import { getWalletByName } from './wallet'
+import { gql, useMutation, useQuery } from '@apollo/client'
+import { useMe } from './me'
+
+const FollowCheckbox = ({ value, ...props }) => {
+  const [,, helpers] = useField(props.name)
+
+  useEffect(() => {
+    helpers.setValue(value)
+  }, [value])
+
+  return (
+    <Checkbox {...props} />
+  )
+}
+
+export function WalletLogs ({ wallet, embedded }) {
+  const logs = useWalletLogs(wallet)
+
+  const router = useRouter()
+  const { follow: defaultFollow } = router.query
+  const [follow, setFollow] = useState(defaultFollow ?? true)
+  const tableRef = useRef()
+  const scrollY = useRef()
+  const showModal = useShowModal()
+
+  useEffect(() => {
+    if (follow) {
+      tableRef.current?.scroll({ top: tableRef.current.scrollHeight, behavior: 'smooth' })
+    }
+  }, [logs, follow])
+
+  useEffect(() => {
+    function onScroll (e) {
+      const y = e.target.scrollTop
+
+      const down = y - scrollY.current >= -1
+      if (!!scrollY.current && !down) {
+        setFollow(false)
+      }
+
+      const maxY = e.target.scrollHeight - e.target.clientHeight
+      const dY = maxY - y
+      const isBottom = dY >= -1 && dY <= 1
+      if (isBottom) {
+        setFollow(true)
+      }
+
+      scrollY.current = y
+    }
+    tableRef.current?.addEventListener('scroll', onScroll)
+    return () => tableRef.current?.removeEventListener('scroll', onScroll)
+  }, [])
```

Comment:
I know it's not the subject of this PR, but I've been wondering if we might avoid the scroll by sorting in descending order - that is, most recent entries first. Our `recent` sort is works this way as do most UI's on the internet with the exception of chat apps.

Reply:
Mhh, I remember considering this. I think I thought there would be an expectation for logs to work like printing line by line and thus scroll downwards. But I think we can indeed avoid this follow functionality by simply reversing the order.

Reply:
Done in 2ee983d0